jena-core/src/main/java/org/apache/jena/graph/impl/TransactionHandlerBase.java

```diff
@@ -36,21 +40,45 @@ public TransactionHandlerBase() {
      */
     @Override
     public Object executeInTransaction(Command c) {
+        return executeRtn( () -> c.execute() ) ;
+    }
+
+    /* Abort but don't let problems with the transaction system itself cause loss of the exception */ 
+    private void abort(Throwable e) {
+        try { abort() ; }
+        catch (Throwable th) { e.addSuppressed(th); }
+    }
+    
+    /**
+     * Execute the runnable <code>action</code> within a transaction. If it completes normally,
+     * commit the transaction, otherwise abort the transaction.
+     */
+    @Override
+    public void execute( Runnable action ) {
         begin() ;
-        Object result ;
         try {
-            result = c.execute() ;
+            action.run();
             commit() ;
-            return result ;
         }
         catch (JenaException e) { abort(e) ; throw e ; }
         catch (Throwable e)     { abort(e) ; throw new JenaException(e) ; }
     }
-
-    /* Abort but don't let problems with the transaction system itself cause loss of the exception */ 
-    private void abort(Throwable e) {
-        try { abort() ; }
-        catch (Throwable th) { e.addSuppressed(th); }
+    
+    /**
+     * Execute the supplier <code>action</code> within a transaction. If it completes normally,
+     * commit the transaction and return the result, otherwise abort the transaction.
+     */
+    @Override
+    public <T> T executeRtn( Supplier<T> action ) {
+        begin() ;
+        T result ;
```

Comment:
I'm just wondering why this isn't 
```
try {
    T result = action.get() ;
    commit() ;
    return result ;
}
```
i.e. what is the need to declare `result` outside of the `try`?